src/test/java/org/openstreetmap/atlas/utilities/conversion/StringToPredicateConverterTest.java

```diff
@@ -52,4 +118,84 @@ public void testConversion()
         Assert.assertTrue(predicate6.test(5));
     }
+
+    @Test
+    public void testConvertUnsafe()
+    {
+        final Predicate<String> predicate1 = new StringToPredicateConverter<String>()
+                .convertUnsafe("e.equals(\"foo\")");
+        final Predicate<Map<String, String>> predicate2 = new StringToPredicateConverter<Map<String, String>>()
+                .convertUnsafe("e.containsKey(\"foo\")");
+        final Predicate<Integer> predicate3 = new StringToPredicateConverter<Integer>()
+                .convertUnsafe("e == 3");
+        final Predicate<AtlasEntity> predicate4 = new StringToPredicateConverter<AtlasEntity>()
+                .convertUnsafe("e.getTags().containsKey(\"mat\")");
+        final Predicate<Integer> predicate5 = new StringToPredicateConverter<Integer>()
+                .withAddedStarImportPackages("org.openstreetmap.atlas.utilities.random")
+                .convertUnsafe("e.intValue() == RandomTagsSupplier.randomTags(5).size()");
+        final Predicate<Integer> predicate6 = new StringToPredicateConverter<Integer>()
+                .withAddedStarImportPackages(
+                        Collections.singletonList("org.openstreetmap.atlas.utilities.random"))
+                .convertUnsafe("e.intValue() == RandomTagsSupplier.randomTags(5).size()");
+
+        Assert.assertTrue(predicate1.test("foo"));
+        Assert.assertFalse(predicate1.test("bar"));
+
+        Assert.assertTrue(predicate2.test(Maps.hashMap("foo", "bar", "baz", "bat")));
+        Assert.assertFalse(predicate2.test(Maps.hashMap("baz", "bat", "bar", "mat")));
+
+        Assert.assertTrue(predicate3.test(3));
+        Assert.assertFalse(predicate3.test(10));
+
+        Assert.assertTrue(predicate4.test(this.rule.getAtlas().point(3)));
+        Assert.assertFalse(predicate4.test(this.rule.getAtlas().point(1)));
+
+        Assert.assertTrue(predicate5.test(5));
+        Assert.assertTrue(predicate6.test(5));
+    }
+
+    @Test
+    public void testImportInjectionProtection1()
+    {
+        this.expectedException.expect(CoreException.class);
+        this.expectedException.expectMessage("Invalid import");
+        new StringToPredicateConverter<Integer>()
+                .withAddedStarImportPackages(Collections.singletonList(
+                        "org.openstreetmap.atlas.utilities.random.*;System.out.println(\"INJECTED\");import org.openstreetmap.atlas"))
+                .convertUnsafe("e.intValue() == RandomTagsSupplier.randomTags(5).size()");
+    }
+
+    @Test
+    public void testImportInjectionProtection2()
+    {
+        this.expectedException.expect(CoreException.class);
+        this.expectedException.expectMessage("Invalid import");
+        new StringToPredicateConverter<Integer>()
+                .withAddedStarImportPackages(Collections.singletonList("System.out.println()"))
+                .convertUnsafe("e.intValue() == RandomTagsSupplier.randomTags(5).size()");
+    }
+
+    private void speedTest()
```

Comment:
Maybe invoke this one in a main method so it can easily be run locally?

Reply:
Whoops I actually meant to remove this! Do you think it's something we should leave in? Otherwise I'll just remove it.

Reply:
Sure, removing it is fine if you're ok with that!